core/src/avm2/error.rs

```diff
@@ -20,17 +20,22 @@ pub enum Error<'gc> {
     RustError(Box<dyn std::error::Error>),
 }
 
-impl<'gc> Error<'gc> {
-    pub fn detailed_message(&self, activation: &mut Activation<'_, 'gc>) -> String {
+impl<'gc> Debug for Error<'gc> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         if let Error::AvmError(error) = self {
             if let Some(error) = error.as_object().and_then(|obj| obj.as_error_object()) {
-                if let Ok(text) = error.display_full(activation) {
-                    return text.to_string();
-                }
+                return write!(
+                    f,
+                    "{}",
+                    error.display_full().expect("Failed to display error")
```

Comment:
This should never be hit, considering how simple `display_full` is. I think it's better to panic if that ever happens, since something has gone very wrong (e.g. the slotids have changed).